Shared/Samples/Show line of sight between geoelements/ShowLineOfSightBetweenGeoelementsView.swift

```diff
@@ -0,0 +1,357 @@
+            ) {
+                (taxiGraphic.symbol as? ModelSceneSymbol)?.heading = Float(distance.azimuth1.value)
+            }
+            setVisibilityStatus()
+        }
+        
+        /// Updates the UI visibility status based on the result of the line of sight analysis.
+        private func setVisibilityStatus() {
+            guard let lineOfSight else { return }
+            switch lineOfSight.targetVisibility {
+            case .obstructed:
+                visibilityStatus = "Obstructed"
+                taxiGraphic?.isSelected = false
+            case .visible:
+                visibilityStatus = "Visible"
+                taxiGraphic?.isVisible = true
+            case .unknown:
+                visibilityStatus = "Unknown"
+                taxiGraphic?.isSelected = false
+            @unknown default:
+                visibilityStatus = "Unknown Status"
+            }
+        }
+        
+        /// Returns a point interpolated between two coordinates based on a progress ratio.
+        /// - Parameters:
+        ///   - from: Start point.
+        ///   - to: End point.
+        ///   - progress: A value representing interpolation progress.
+        private func interpolatedPoint(from: Point, to: Point, progress: Double) -> Point {
+            let x = from.x + (to.x - from.x) * progress
+            let y = from.y + (to.y - from.y) * progress
+            return Point(x: x, y: y, spatialReference: .wgs84)
+        }
+        
+        /// Updates the Z (height) value of the observer's point geometry.
+        /// - Parameter height: The new observer height.
+        private func changeObserverHeight(_ height: Double) {
+            guard let observer = observerGraphic,
+                  let geometry = observer.geometry as? Point else { return }
+            // Create a new point with the updated Z (height) value.
+            let updatedPoint = Point(
+                x: geometry.x,
+                y: geometry.y,
+                z: height,
+                spatialReference: geometry.spatialReference
+            )
+            
+            // Update the observer's geometry.
+            observer.geometry = updatedPoint
+        }
+    }
+}
+
+extension URL {
+    // URL of the elevation service - provides elevation component of the scene
+    static var elevationService: URL {
+        URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+    }
+    
+    // URL of the building service - provides builiding models
+    static var buildingsService: URL {
+        URL(string: "https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/Buildings_NewYork_v18/SceneServer/layers/0")!
+    }
+    
+    /// A URL to the loca taxi model files.
+    static var taxi: URL {
+        Bundle.main.url(forResource: "dolmus", withExtension: "3ds", subdirectory: "Dolmus3ds")!
+    }
+}
+
+#Preview {
+    ShowLineOfSightBetweenGeoelementsView()
+}
```

Comment:
```suggestion
```
Unfortunately, ODR doesn't support SwiftUI preview yet (#315), so the preview will always crash when we used an ODR.